teamup-server/src/main/java/com/project/teamup/controller/UserController.java

```diff
@@ -19,10 +19,27 @@
     private UserService userService;
     @Autowired
     private UserMapper userMapper;
+    @Autowired
+    private EmailService emailService;
 
     @PostMapping(value = "/save")
-    public UserDTO saveUser(@RequestBody UserDTO user) {
-        return userMapper.toDto(userService.save(userMapper.toEntity(user)));
+    public ResponseEntity<UserDTO> saveUser(@RequestBody UserDTO user,@RequestBody String token) {
+        if (userService.isValidToken(token)) {
+            return ResponseEntity.ok(userMapper.toDto(userService.save(userMapper.toEntity(user))));
+        }
+
+        return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+    }
+    @PostMapping(value = "/generateToken/{id}")
+    public ResponseEntity generateToken(@PathVariable Long id){
+        //User userEntity = userMapper.toEntity(user);
```

Comment:
If you want to keep the @PathVariable, please change the 'id' to 'userId' and delete the commented line